sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,377 @@
+     * parent. This means essentially it doesn't matter what JsonElement is
+     * passed in to the add method, all it has to do is pass the JsonWriter to
+     * the serialize() method of whatever JsonElement it is using, and it will
+     * serialize.
+     *
+     * @param jsonWriter the JsonWriter to use
+     * @return the JsonWriter
+     * @throws IOException if the JsonWriter throws an exception
+     */
+    public JsonWriter serialize(JsonWriter jsonWriter) throws IOException {
+        // Start writing the array into jsonWriter, if it hits another array or
+        // object, then pass the writer in to that objects serialize (which will
+        // call this method again) and then return the writer to here. This will
+        // unnest the lot.
+
+        jsonWriter.writeStartArray();
+
+        for (JsonElement element : elements) {
+            element.serialize(jsonWriter);
+        }
+
+        jsonWriter.writeEndArray();
+        return jsonWriter;
+    }
+
+//    public String toJson() {
+//        // String reference that will store the resulting JSON string output to
+//        // be returned by toJson
+//        String jsonOutput = "";
+//
+//        // Iterating over all JsonElement objects in the elements JsonArray object,
+//        // appending each JsonElement to jsonOutput
+//        for(Iterator<JsonElement> itr = this.elements.iterator(); itr.hasNext();) {
+//            // Get the next JsonElement from the elements JsonArray object
+//            JsonElement element = itr.next();
+//
+//            // Case: element is a JsonObject, therefore the whole JSON object can
+//            // be stringified via its toJson method, then appended to jsonOutput
+//            if(element instanceof JsonObject) {
+//                jsonOutput += ((JsonObject)element).toJson();
+//            }
+//            // Case: element is a JsonArray, therefore the nested JSON array can
+//            // be stringified via recursively calling toJson and then appending
+//            // the returned String to jsonOutput
+//            else if(element instanceof JsonArray) {
+//                jsonOutput += ((JsonArray)element).toJson();
+//            }
+//            // Case: element is not a JsonObject or JsonArray, therefore it can
+//            // be simply converted to a String object and appended to jsonOutput
+//            else { jsonOutput += element; }
+//
+//            // Case: haven't reached the end of the JSON array, therefore more
+//            // elements must be converted to String objects and appended to the
+//            // jsonOutput
+//            // Elements are separated by commas.
+//            if(itr.hasNext()) { jsonOutput += ", "; }
+//        }
+//
+//        // Returning the resulting String representation of the JsonArray object
+//        // JSON arrays are delimited by opening and closing square brackets.
+//        return "[" + jsonOutput + "]";
+//    }
+
```

Comment:
Start pruning out unused code like this.

Reply:
@JonathanGiles done